### PR TITLE
Aggregations and faster video

### LINKED ISSUE #634: [BUG] Field vs expression issue with SortBy view stage
If a `SortBy` stage contains an expression, its display text should show `{...}`, and clicking into the field should open the JSON editor.

(Just like what happens if you click into the filter expression in the `FilterDetections` stage in the App in the example below)

```py
import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

dataset = foz.load_zoo_dataset("coco-2017", split="validation")

person_view = (
    dataset
    .filter_detections("ground_truth", F("label") == "person")
    .sort_by(F("ground_truth.detections").length(), reverse=True)
)

# Show the view in the App
session = fo.launch_app(view=person_view)
```

Screen Shot 2020-10-26 at 11 40 36 AM

### LINKED ISSUE #639: [FR] Revert view stage fields to "compact mode" when user clicks out of them
View stage fields should revert to their "compact display mode" when the user clicks out of them.

For example, the field of the `Select` stage below has a nice compact display before I click into it; currently, I cannot get it to collapse when clicking out of it, which makes it hard to read the remaining stages.

Some fields do revert, like the filter field of the `FilterDetections()` stage; it always collapses to `{...}` when the user clicks out of it; which is expected and nice.

**Before clicking in**

Screen Shot 2020-10-26 at 1 31 47 PM

**While active**

Screen Shot 2020-10-26 at 1 31 56 PM

**After clicking out** (desired behavior is reverting to "before clicking in" state)

Screen Shot 2020-10-26 at 1 31 52 PM

### LINKED ISSUE #640: [FR] Updating the dataset/view should refresh expanded sample modal
If `session.dataset` or `session.view` is updated when the expanded sample view is open in the App, the modal needs to refresh. Currently it does not.

If the previously visible sample is no longer in the dataset/view, then the modal should just close.

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Resolves #640
Resolves #639
Resolves #634

* Introduces FiftyOne Aggregations
* Full labels graph support for all label types on both frames and samples
* Display options parity for labels, scalars and tags are a WIP still
* Faster video

## How is this patch tested? If it is not, please explain why.

Aggregation unit tests.

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

TBD

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [x] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: docs/source/user_guide/using_aggregations.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+more than one thing about a dataset or view.
+
+.. code-block:: python
+    :linenos:
+    import fiftyone as fo
+
+    aggregations = [fo.CountLabels("predictions"), fo.ConfidenceBounds("predictions")]
+    dataset = fo.load_dataset("my_dataset")
+    view = datatset.sort_by("uniqueness", reverse=True).limit(10)
+    for result in aggregations:
```
Comment (brimoor) [inline docs/source/user_guide/using_aggregations.rst:49]: Isn't this missing a connection to `view`? Maybe this:

```py
for result in view.aggregate(aggregations):
    print(result)
```

Comment (benjaminpkane) [reply docs/source/user_guide/using_aggregations.rst:49]: Yes, thanks!

File: fiftyone/core/dataset.py [hunk 2]
```diff
@@ -204,7 +206,7 @@ def __init__(self, name=None, persistent=False, _create=True):
         self._deleted = False
 
     def __len__(self):
-        return self._sample_collection.count_documents({})
+        return self.aggregate(foa.Count()).count
```
Comment (brimoor) [inline fiftyone/core/dataset.py:209]: How about leveraging the power of aggregations by computing `len(self)` and `get_tags()` via a single `aggregate()` call in the definitions of `Dataset.summary()` and `DatasetView.summary()`? 😄

Comment (benjaminpkane) [reply fiftyone/core/dataset.py:209]: Of course, on it.

**Review by brimoor — APPROVED** [review]
I haven't fully tested this (and not sure if you're fully done), but I read the code and it LGTM

**Pull request merged**